optimum/onnx/configuration.py

```diff
@@ -0,0 +1,229 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import copy
+import dataclasses
+import warnings
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from typing import TYPE_CHECKING, Any, Callable, Dict, Iterable, List, Mapping, Optional, Tuple, Union
+
+import numpy as np
+from packaging import version
+from transformers.onnx.utils import (
+    ParameterFormat,
+    compute_effective_axis_dimension,
+    compute_serialized_parameters_size,
+)
+from transformers.utils import TensorType, is_torch_available, is_vision_available, logging
+
+
+if TYPE_CHECKING:
+    from transformers.configuration_utils import PretrainedConfig
+    from transformers.feature_extraction_utils import FeatureExtractionMixin
+    from transformers.tokenization_utils_base import PreTrainedTokenizerBase
+
+import torch
+from transformers.onnx import OnnxConfig, OnnxConfigWithPast, OnnxSeq2SeqConfigWithPast
+
+
+logger = logging.get_logger(__name__)
+
+
+class OnnxConfigWithLoss(OnnxConfig, ABC):
+    """
+    Wrapper for the childern classes of `transformers.onnx.OnnxConfig` to export the model through the ONNX format with loss in outputs.
+    """
+
+    _tasks_to_extra_inputs = {
+        "default": OrderedDict({"labels": {0: "batch"}}),
+        "masked-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "causal-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "seq2seq-lm": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "sequence-classification": OrderedDict({"labels": {0: "batch"}}),
+        "token-classification": OrderedDict({"labels": {0: "batch", 1: "sequence"}}),
+        "multiple-choice": OrderedDict({"labels": {0: "batch"}}),
+        "question-answering": OrderedDict(
+            {
+                "start_positions": {0: "batch"},
+                "end_positions": {0: "batch"},
+            }
+        ),
+        "image-classification": OrderedDict({"labels": {0: "batch"}}),
+    }
+    _tasks_to_extra_outputs = {
+        "default": OrderedDict({"loss": {}}),
+    }
+
+    def __init__(self, config: OnnxConfig):
+        self.__dict__ = config.__dict__
```

Comment:
Yes, I feel like it would be better to only copy the task then, or at least to work on a deepcopy of the original config.
```suggestion
        self.__dict__ = copy.deepcopy(config.__dict__)
```